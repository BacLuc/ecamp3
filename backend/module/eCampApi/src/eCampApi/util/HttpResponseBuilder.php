<?php

namespace eCampApi\util;

use Laminas\Http\Response;

class HttpResponseBuilder {
    private int $httpStatusCode;

    public function __construct(int $httpStatusCode) {
        $this->httpStatusCode = $httpStatusCode;
    }

    public static function ok(): HttpResponseBuilder {
        return new HttpResponseBuilder(Response::STATUS_CODE_200);
    }

    public function build(): Response {
        $response = new Response();
        $response->setStatusCode($this->httpStatusCode);

        return $response;
    }
}
