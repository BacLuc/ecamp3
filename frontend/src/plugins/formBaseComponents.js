// Global registration of all Vue components in folder components/form/base

// import upperFirst from 'lodash/upperFirst'
// import camelCase from 'lodash/camelCase'

import ECheckbox from '@/components/form/base/ECheckbox.vue'
import EColorPicker from '@/components/form/base/EColorPicker.vue'
import EDatePicker from '@/components/form/base/EDatePicker.vue'
import ERichtext from '@/components/form/base/ERichtext.vue'
import ESelect from '@/components/form/base/ESelect.vue'
import ESwitch from '@/components/form/base/ESwitch.vue'
import ETextarea from '@/components/form/base/ETextarea.vue'
import ETextField from '@/components/form/base/ETextField.vue'
import ETimePicker from '@/components/form/base/ETimePicker.vue'

/*
const requireComponent = require.context(
  // The relative path of the components folder
  '../components/form/base',
  // Whether or not to look in subfolders
  false,
  // The regular expression used to match base component filenames
  /[A-Z]\w+\.(vue|js)$/
) */

class FormBaseComponentsPlugin {
  install (Vue, options) {
    Vue.component('ECheckbox', ECheckbox)
    Vue.component('EColorPicker', EColorPicker)
    Vue.component('EDatePicker', EDatePicker)
    Vue.component('ERichtext', ERichtext)
    Vue.component('ESelect', ESelect)
    Vue.component('ESwitch', ESwitch)
    Vue.component('ETextarea', ETextarea)
    Vue.component('ETextField', ETextField)
    Vue.component('ETimePicker', ETimePicker)

    /*
    requireComponent.keys().forEach(fileName => {
      // Get component config
      const componentConfig = requireComponent(fileName)

      // Get PascalCase name of component
      const componentName = upperFirst(
        camelCase(
          // Gets the file name regardless of folder depth
          fileName
            .split('/')
            .pop()
            .replace(/\.\w+$/, '')
        )
      )

      // Register component globally
      Vue.component(
        componentName,
        // Look for the component options on `.default`, which will
        // exist if the component was exported with `export default`,
        // otherwise fall back to module's root.
        componentConfig.default || componentConfig
      )
    }) */
  }
}

export default new FormBaseComponentsPlugin()
