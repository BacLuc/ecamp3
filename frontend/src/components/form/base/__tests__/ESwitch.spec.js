import Vue from 'vue'
import Vuetify from 'vuetify'

import formBaseComponents from '@/plugins/formBaseComponents'

import { mount as mountComponent } from '@vue/test-utils'
import ESwitch from '@/components/form/base/ESwitch'
import { touch } from '@/test/util'

Vue.use(Vuetify)
Vue.use(formBaseComponents)

describe('An ESwitch', () => {
  let vuetify

  const mount = (options) => mountComponent(ESwitch, { vuetify, ...options })
  beforeEach(() => {
    vuetify = new Vuetify()
  })
  test('looks like a switch', async () => {
    const wrapper = mount()
    await wrapper.setProps({ vModel: false })
    expect(wrapper.element).toMatchSnapshot('unchecked')

    await wrapper.setProps({ vModel: true })
    expect(wrapper.element).toMatchSnapshot('checked')
  })

  test('is checked when initialized checked', () => {
    const wrapper = mount({
      propsData: {
        vModel: true
      }
    })
    expect(wrapper.find('input[type=checkbox]').element.getAttribute('vmodel')).toBe('true')
  })

  test('updates switch when vModel changes', async () => {
    const wrapper = mount()
    expect(wrapper.find('input[type=checkbox]').element.getAttribute('vmodel')).toBeNull()

    await wrapper.setProps({ vModel: true })
    expect(wrapper.find('input[type=checkbox]').element.getAttribute('vmodel')).toBe('true')

    await wrapper.setProps({ vModel: false })
    expect(wrapper.find('input[type=checkbox]').element.getAttribute('vmodel')).toBeNull()
  })

  test('updates vModel when user toggles with click', async () => {
    const wrapper = mount()
    const input = wrapper.find('input')

    const change = jest.fn()
    wrapper.vm.$on('input', (event) => change(event))

    await input.trigger('click')
    expect(change).toBeCalledTimes(1)
    expect(change).toBeCalledWith(true)

    jest.resetAllMocks()
    await input.trigger('click')
    expect(change).toBeCalledTimes(1)
    expect(change).toBeCalledWith(false)
  })

  test('updates vModel when user toggles with touch swipe', async () => {
    const wrapper = mount()

    const change = jest.fn()
    wrapper.vm.$on('input', (event) => change(event))

    touch(wrapper.find('.v-input--selection-controls__ripple')).start(0, 0).end(20, 0)
    expect(change).toBeCalledWith(true)

    jest.resetAllMocks()
    touch(wrapper.find('.v-input--selection-controls__ripple')).start(0, 0).end(-20, 0)
    expect(change).toBeCalledWith(false)
  })

  test('updates vModel when user toggles with keys', async () => {
    const wrapper = mount()
    const input = wrapper.find('input')

    const change = jest.fn()
    wrapper.vm.$on('input', (event) => change(event))

    input.trigger('keydown.right')
    expect(change).toHaveBeenCalledWith(true)
    expect(change).toHaveBeenCalledTimes(1)

    input.trigger('keydown.right')
    expect(change).toHaveBeenCalledTimes(1)

    jest.resetAllMocks()
    input.trigger('keydown.left')
    expect(change).toHaveBeenCalledWith(false)
    expect(change).toHaveBeenCalledTimes(1)
  })
})
