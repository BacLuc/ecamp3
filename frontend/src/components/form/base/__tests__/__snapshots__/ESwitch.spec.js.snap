// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`An ESwitch looks like a switch: checked 1`] = `
<span>
  <div
    class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--switch v-input--switch--inset"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-input--selection-controls__input"
        >
          <input
            aria-checked="false"
            aria-disabled="false"
            id="input-4"
            role="switch"
            type="checkbox"
            value=""
            vmodel="true"
          />
          <div
            class="v-input--selection-controls__ripple"
          />
          <div
            class="v-input--switch__track theme--light"
          />
          <div
            class="v-input--switch__thumb theme--light"
          >
            <transition-stub
              mode="out-in"
              name="fab-transition"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</span>
`;

exports[`An ESwitch looks like a switch: unchecked 1`] = `
<span>
  <div
    class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--switch v-input--switch--inset"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-input--selection-controls__input"
        >
          <input
            aria-checked="false"
            aria-disabled="false"
            id="input-4"
            role="switch"
            type="checkbox"
            value=""
          />
          <div
            class="v-input--selection-controls__ripple"
          />
          <div
            class="v-input--switch__track theme--light"
          />
          <div
            class="v-input--switch__thumb theme--light"
          >
            <transition-stub
              mode="out-in"
              name="fab-transition"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</span>
`;
